data/list/basic.lean

```diff
@@ -2554,6 +2556,11 @@ by simp [diff_eq_foldl]
   map f (l₁.diff l₂) = (map f l₁).diff (map f l₂) :=
 by simp [diff_eq_foldl, map_foldl_erase finj]
 
+theorem diff_sublist : ∀ l₁ l₂ : list α, l₁.diff l₂ <+ l₁
+| l₁ []      := by simp
+| l₁ (a::l₂) := calc l₁.diff (a :: l₂) = (l₁.erase a).diff l₂ : diff_cons _ _ _
+  ... <+ l₁.erase a : diff_sublist _ _
+  ... <+ l₁ : list.erase_sublist _ _
```

Comment:
I'm not sure what @digama0 thinks of it but I prefer aligning the `:` hints of the `calc` and aligning the `=` and `<+`

Reply:
I agree. It's a bit inconvenient that the operators have different sizes; I will leave it to your discretion whether to add an additional space after the `=` to make the RHS also line up, but I agree that the `=` should line up with the `<+`. You could achieve this either with a large indent on the last two lines, or by breaking the first line after `calc` or after the LHS so that the indent isn't so large.